### ISSUE TITLE
Failed to pull singularity image viroprofiler-base:v0.2.2, manifest unknown

### ISSUE THREAD

**Issue by liuronghyw** [deng-lab/viroprofiler#7]
### Description of the bug

Dear Dr. Long,
I have tried to install viroprofile for about two weeks, but still failed. When I install the newest version of viroprofiler with the following command:
nextflow run deng-lab/viroprofiler -r main -profile singularity --mode "setup" --db "/public/home/hpc4011032/db/viroprofiler"

The error:

N E X T F L O W  ~  version 22.10.6
WARNING: Could not load nf-core/config profiles: https://raw.githubusercontent.com/nf-core/configs/master/nfcore_custom.config
Launching `https://github.com/deng-lab/viroprofiler` [infallible_khorana] DSL2 - revision: 222364d5b7 [main]

----------------------------------------------------------------------------------------------------------------
                                               __                                                             
 oooooo     oooo  o8o                        88  88                        .o88o.  o8o  oooo                  
  `888.     .8'   `"'                       88 ss 88                       888 `"  `"'  `888                  
   `888.   .8'   oooo  oooo d8b  .ooooo.     88__88    oooo d8b  .ooooo.  o888oo  oooo   888   .ooooo.  oooo d8b
    `888. .8'    `888  `888""8P d88' `88b      ||      `888""8P d88' `88b  888    `888   888  d88' `88b `888""8P
     `888.8'      888   888     888   888     _||_      888     888   888  888     888   888  888ooo888  888  
      `888'       888   888     888   888   // || \\    888     888   888  888     888   888  888    .o  888  
       `8'       o888o d888b    `Y8bod8P'  //      \\  d888b    `Y8bod8P' o888o   o888o o888o `Y8bod8P' d888b 
  ViroProfiler v0.2
----------------------------------------------------------------------------------------------------------------
Core Nextflow options
  revision       : main
  runName        : infallible_khorana
  containerEngine: singularity
  launchDir      : /public/home/hpc4011032/ICB_viro/viroprofiler
  workDir        : /public/home/hpc4011032/ICB_viro/viroprofiler/work
  projectDir     : /public/home/hpc4011032/.nextflow/assets/deng-lab/viroprofiler
  userName       : hpc4011032
  profile        : singularity
  configFiles    : /public/home/hpc4011032/.nextflow/assets/deng-lab/viroprofiler/nextflow.config

Input/output options
  mode           : setup
  db             : /public/home/hpc4011032/db/viroprofiler
  outdir         : output

Others
  use_iphop      : true
  use_dram       : true

Max job request options
  max_cpus       : 1
  max_memory     : 8.GB
  max_time       : 12.h

!! Only displaying parameters that differ from the pipeline defaults !!
----------------------------------------------------------------------------------------------------------------
If you use ViroProfiler for your analysis please cite:

* The ViroProfiler pipeline
  https://doi.org/10.1080/19490976.2023.2192522

* The nf-core framework
  https://doi.org/10.1038/s41587-020-0439-x

* Software dependencies
  https://github.com/deng-lab/viroprofiler/blob/main/CITATIONS.md
----------------------------------------------------------------------------------------------------------------
[-        ] process > VIROPROFILER:SETUP:DB_CHECKV     -
[-        ] process > VIROPROFILER:SETUP:DB_VIBRANT    [  0%] 0 of 1
executor >  local (3)
[-        ] process > VIROPROFILER:SETUP:DB_CHECKV     -
executor >  local (3)
[-        ] process > VIROPROFILER:SETUP:DB_CHECKV     -
executor >  local (3)
[-        ] process > VIROPROFILER:SETUP:DB_CHECKV     -
executor >  local (3)
[-        ] process > VIROPROFILER:SETUP:DB_CHECKV     -
[ec/d12e50] process > VIROPROFILER:SETUP:DB_VIBRANT    [100%] 1 of 1 ✔
[-        ] process > VIROPROFILER:SETUP:DB_VIRSORTER2 -
[61/71fda0] process > VIROPROFILER:SETUP:DB_DRAM       [100%] 1 of 1 ✔
[b1/56b5ad] process > VIROPROFILER:SETUP:DB_IPHOP      [100%] 1 of 1 ✔
executor >  local (3)
[-        ] process > VIROPROFILER:SETUP:DB_CHECKV     -
[ec/d12e50] process > VIROPROFILER:SETUP:DB_VIBRANT    [100%] 1 of 1 ✔
[-        ] process > VIROPROFILER:SETUP:DB_VIRSORTER2 -
[61/71fda0] process > VIROPROFILER:SETUP:DB_DRAM       [100%] 1 of 1 ✔
[b1/56b5ad] process > VIROPROFILER:SETUP:DB_IPHOP      [100%] 1 of 1 ✔
[-        ] process > VIROPROFILER:SETUP:DB_VREFSEQ    -
Pulling Singularity image docker://denglab/viroprofiler-base:v0.2.2 [cache /public/home/hpc4011032/ICB_viro/viroprofiler/./temp/denglab-viroprofiler-base-v0.2.2.img]
Pulling Singularity image docker://denglab/viroprofiler-base:v0.2 [cache /public/home/hpc4011032/ICB_viro/viroprofiler/./temp/denglab-viroprofiler-base-v0.2.img]
Error executing process > 'VIROPROFILER:SETUP:DB_VIRSORTER2'

Caused by:
  Failed to pull singularity image
  command: singularity pull  --name denglab-viroprofiler-base-v0.2.2.img.pulling.1682469314842 docker://denglab/viroprofiler-base:v0.2.2 > /dev/null
  status : 255
  message:
    FATAL:   While making image from oci registry: error fetching image to cache: failed to get checksum for docker://denglab/viroprofiler-base:v0.2.2: Error reading manifest v0.2.2 in docker.io/denglab/viroprofiler-base: manifest unknown: manifest unknown

Would you please help me about this issue ? 

### Command used and terminal output

```console
nextflow run deng-lab/viroprofiler -r main -profile singularity --mode "setup" --db "/public/home/hpc4011032/db/viroprofiler"
```

### Relevant files

![报错信息](https://user-images.githubusercontent.com/42331632/234438694-12d2e695-b64f-44e4-b373-7a496a077b4c.png)

### System information

N E X T F L O W  ~  version 22.10.6

HPC

slurm

Singularity

CentOS Linux

viroprofiler 0.2.2

**Comment by XC-Zhai** [comment]
I had the same issue, you can run: 
nextflow run deng-lab/viroprofiler -r fix_iphop -profile singularity --mode "setup" ,
would set up most steps.

Best,
Xichuan

**Comment by rujinlong** [comment]
This bug is fixed in v0.2.3. Please update the pipeline using `nextflow pull deng-lab/viroprofiler` and rerun using parameter `-r main` or `-r v0.2.3`.
